lib/mix/lib/mix/scm/git.ex

```diff
@@ -126,17 +126,17 @@ defmodule Mix.SCM.Git do
     update_origin(opts[:git])
 
     # Fetch external data
-    branch_or_tag = opts[:branch] || opts[:tag]
+    rev = get_lock_rev(opts[:lock], opts) || get_opts_rev(opts)
```

Comment:
This is a change in this PR where we are always fetching a locked revision. It wasn't like this before. Are we sure we want to go down this path?

Reply:
It was like this before 786f9592b7d73a4b1a3c8e6612e51cdc924dab9f. It's my understanding that `mix.lock` stores the full SHA1, so it should be okay to fetch it. Do you anticipate some problem?

Reply:
Ah, perfect. Let's give it a try like this once again then!